keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/ArtifactEvent.kt

```diff
@@ -8,6 +9,10 @@ data class ArtifactEvent(
   val details: Map<String, Any>?
 )
 
+/**
+ * This event optionally takes [statuses] so that we can fetch the most relevant first artifact.
+ */
 data class ArtifactRegisteredEvent(
-  val artifact: DeliveryArtifact
+  val artifact: DeliveryArtifact,
+  val statuses: List<ArtifactStatus> = enumValues<ArtifactStatus>().toList()
```

Comment:
I don't understand why the statuses are needed here

Reply:
It's in the class comment. For more information, when we register the artifact we fetch the latest version. I want to make sure we fetch the latest version with the correct status (or a status we care about) so we can deploy right away if needed.

Reply:
How do we know what statuses we'd want to consider in this scenario?

Reply:
We publish the artifact register event in the `ImageHandler`, so we know what statuses we need to consider (it's in the `ImageSpec` now).

Reply:
Okay, I see, so this is just to ensure that the new resource gets an image it can use.

Reply:
Exactly.